### PR TITLE
feat: Show alert in the expense form if category is mandatory and no category is allowed

### PR DESCRIPTION
**Pull request by arjunaj5**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=ZRXpBcE)
https://app.clickup.com/t/85ztzt5nt

Screenshot 2023-09-27 at 11 09 05 AM

Similar existing design on the projects is used here.
Screenshot 2023-09-27 at 1 02 21 PM

### <samp>🤖 Generated by Copilot at fb0b1a9</samp>

Improved category selection and validation for adding and editing expenses. Displayed an alert for users without categories.

### <samp>🤖 Generated by Copilot at fb0b1a9</samp>

> _To add or edit an expense_
> _You need a category that makes sense_
> _But if you have none_
> _You'll see an alert run_
> _And tell you to fix this offense_

### <samp>🤖 Generated by Copilot at fb0b1a9</samp>

* Change the condition for showing the category field to only depend on the accounting software connection ([link](https://github.com/fylein/fyle-mobile-app/pull/2456/files?diff=unified&w=0#diff-222fd948c0d0cdfd9422c33d7e78705dddeb3bb00664d65ab64f77331691bf2cL442-R442))
* Add an alert message to inform the user about the mandatory category field and the possible need to contact the admin ([link](https://github.com/fylein/fyle-mobile-app/pull/2456/files?diff=unified&w=0#diff-222fd948c0d0cdfd9422c33d7e78705dddeb3bb00664d65ab64f77331691bf2cR1157-R1165))

## Clickup
Please add link here

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.html [hunk 1]
```diff
@@ -1154,6 +1154,15 @@
...
           </ng-container>
+
+          <ng-container *ngIf="filteredCategories$|async as categories">
+            <app-fy-alert-info
+              *ngIf="txnFields?.org_category_id?.is_mandatory && !categories.length"
+              [message]="'There are no Categories assigned to you, so this expense will be saved as a draft. Please contact admin for further help.'"
+              [type]="'warning'"
+            >
+            </app-fy-alert-info>
+          </ng-container>
```
Comment (arjunaj5) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.html:1165]: This is the only change, rest are lint fixes :(

File: src/app/fyle/add-edit-expense/add-edit-expense.page.html [hunk 2]
```diff
@@ -439,7 +439,7 @@
           </ng-container>
 
           <ng-container *ngIf="txnFields$|async as txnFields">
-            <ng-container *ngIf="!((isProjectsEnabled$|async) && !(isConnected$|async))">
+            <ng-container *ngIf="isConnected$|async">
```
Comment (arjunaj5) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.html:442]: This check of i`sProjectsEnabled$|async` was not needed and it was not harmful as the `*ngIf` was always evaluating as true if  `isConnectec$` was true.

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -457,7 +457,7 @@ export class AddEditExpensePage implements OnInit {
     private orgUserSettingsService: OrgUserSettingsService,
     private storageService: StorageService,
     private launchDarklyService: LaunchDarklyService,
-    private platformHandlerService: PlatformHandlerService,
+    private platformHandlerService: PlatformHandlerService
```
Comment (mvaishnavi) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:460]: no change in this file; only lint fixes.

**Review by mvaishnavi — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 3 above)
Comment (arjunaj5) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:460]: **NO change in this file, these are lint fixes**

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -1923,7 +1923,7 @@ export class AddEditExpensePage implements OnInit {
      * - The user is on creating a new expense or editing a DRAFT expense
      */
     if (doRecentOrgCategoryIdsExist && isCategoryEmpty && (!etxn.tx.id || etxn.tx.state === 'DRAFT')) {
-      const autoFillCategory = recentCategories.length && recentCategories[0];
+      const autoFillCategory = recentCategories?.length && recentCategories[0];
```
Comment (arjunaj5) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1926]: This is the only change in this file.

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**